src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -834,6 +869,18 @@ private bool InferConnectionDetails()
                     ReportSNIError(SNIProviders.INVALID_PROV);
                     return false;
                 }
+
+                if (ConnectionProtocol == DataSource.Protocol.None)
```

Comment:
I really believe that we should keep consistency of what it does in DataSource. As I mentioned before, if DataSource is only responsible for parsing connection string, it should not have code that explicitly assigns TCP value when no protocol in connection string. I believe if DataSource should do parsing and additional calculation on top of parsed value, such as manually assigning TCP as default, then we also should bring SSRP call for finding port to DataSource as well.